### PR TITLE
PROD-30975: Implement "user logs out" event for the EDA

### PR DESCRIPTION
**Pull request by tregismoreira**
> [!CAUTION]
> This PR depends on #4151

## Description
This pull request introduces the new EDA event "user logs out" which is dispatches to the Kafka topic `com.getopensocial.cms.user.logout`.

Here's an example of the generated payload:

```
{
	"specversion": "1.0",
	"id": "0cf6cf17-ca0e-4bdd-b791-b3c77a34d300",
	"source": "/user/logout",
	"type": "com.getopensocial.cms.user.logout",
	"datacontenttype": "application/json",
	"time": "2024-10-24T12:15:59Z",
	"data": {
		"user": {
			"id": "3cd53a66-b86a-41b4-8225-66eed5bb5fe3",
			"created": "2024-10-24T12:15:59",
			"updated": "2024-10-24T12:15:59",
			"status": "active",
			"displayName": "admin",
			"firstName": "",
			"lastName": "",
			"email": "devel@getopensocial.com",
			"roles": [
				"authenticated",
				"administrator"
			],
			"timezone": "UTC",
			"language": "en",
			"address": {
				"label": "",
				"countryCode": "",
				"administrativeArea": "",
				"locality": "",
				"dependentLocality": "",
				"postalCode": "",
				"sortingCode": "",
				"addressLine1": "",
				"addressLine2": ""
			},
			"phone": "",
			"function": "",
			"organization": "",
			"href": {
				"canonical": "http://cablecar.localhost/user/1/home"
			}
		},
		"actor": {
			"application": null,
			"user": {
				"id": "3cd53a66-b86a-41b4-8225-66eed5bb5fe3",
				"displayName": "admin",
				"href": {
					"canonical": "http://cablecar.localhost/user/1/home"
				}
			}
		}
	}
}
```

## Release notes (to customers)
N/A

## Issue tracker
https://getopensocial.atlassian.net/browse/PROD-30975

## Theme issue tracker
N/A

## How to test
- [ ] Checkout branch `issue/PROD-30975-eda-user-logout`
- [ ] Enable modules `social_eda` and `social_eda_dispatcher`
- [ ] Log out any user

The message should be dispatched to Kafka. If you are running locally using the Docker containers, you should be able to see the message on the Kafka UI at http://localhost:8080/.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4152-zx0kog9m594jybd6tvvrdzqp2ighyfnm.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6723853e6ec433609ae8aa99

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4152-zx0kog9m594jybd6tvvrdzqp2ighyfnm.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6723853e6ec433609ae8aa99

File: modules/social_features/social_user/asyncapi.yml [hunk 1]
```diff
@@ -404,6 +404,141 @@ channels:
...
+                      nullable: true
+                      description: The first name of the user.
+                    lastName:
+                      type: string
+                      nullable: true
+                      description: The last name of the user.
+                    email:
+                      type: string
+                      format: email
+                      description: The email address of the user.
```
Comment (bramtenhove) [inline modules/social_features/social_user/asyncapi.yml:450]:
```suggestion
```

As part of the payload design we left this out.

File: modules/social_features/social_user/asyncapi.yml [hunk 2]
```diff
@@ -404,6 +404,141 @@ channels:
...
+                      nullable: true
+                      description: The phone number of the user.
+                    function:
+                      type: string
+                      nullable: true
+                      description: The job function of the user.
+                    organization:
+                      type: string
+                      nullable: true
+                      description: The organization of the user.
```
Comment (bramtenhove) [inline modules/social_features/social_user/asyncapi.yml:504]:
```suggestion
```

As part of the payload design we left this out.

**Review by bramtenhove — CHANGES_REQUESTED** [review]
I noticed the payload data contains more than was designed for.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4152-zx0kog9m594jybd6tvvrdzqp2ighyfnm.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6723853e6ec433609ae8aa99

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4152-zx0kog9m594jybd6tvvrdzqp2ighyfnm.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6723853e6ec433609ae8aa99

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4152-zx0kog9m594jybd6tvvrdzqp2ighyfnm.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6723853e6ec433609ae8aa99

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4152-zx0kog9m594jybd6tvvrdzqp2ighyfnm.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6723853e6ec433609ae8aa99

**Review by ribel — APPROVED** [review]
Changes were tested and works as expected.

**Pull request merged**